src/ar_http_iface_cowboy_handler.erl

```diff
@@ -659,21 +665,139 @@ do_handle(Method, [<<"height">>], Req)
 %% @doc If we are given a hash with no specifier (block, tx, etc), assume that
 %% the user is requesting the data from the TX associated with that hash.
 %% Optionally allow a file extension.
-do_handle(<<"GET">>, [<< Hash:43/binary, MaybeExt/binary >>], Req) ->
-	case MaybeExt of
-		<< ".", Part/binary >> ->
-			do_handle(<<"GET">>, [<<"tx">>, Hash, <<"data.", Part/binary>>], Req);
-		<<>> ->
-			do_handle(<<"GET">>, [<<"tx">>, Hash, <<"data.html">>], Req);
-		_ ->
-			not_found(Req)
+handle(<<"GET">>, [<<Hash:43/binary, MaybeExt/binary>>], Req) ->
+	case ar_meta_db:get(apex_domain) of
+		not_found ->
+			handle(<<"GET">>, [<<"tx">>, Hash, <<"data.", MaybeExt/binary>>], Req);
+		ApexDomain ->
+			handle_serve_tx_1(ApexDomain, Hash, MaybeExt, Req)
```

Comment:
We should decode `Hash` before passing it on and call it `TXID` in the next function.